trieste/models/gpflux/interface.py

```diff
@@ -0,0 +1,112 @@
+            batch size 100. Optimizer is required to be a TFOptimizer, attempts to use other
+            optimizers will result in an error.
+        """
+        super().__init__()
+
+        if optimizer is None:
+            optimizer = TFOptimizer(tf.optimizers.Adam(), batch_size=100)
+
+        if not isinstance(optimizer, TFOptimizer):
+            raise ValueError("Optimizer must be a TFOptimizer for GPflux models.")
+
+        if not isinstance(optimizer.optimizer, tf.optimizers.Optimizer):
+            raise ValueError("Model optimizer must be a tf.optimizers.Optimizer for GPflux models.")
+
+        self._optimizer = optimizer
+
+    @property
+    def optimizer(self) -> TFOptimizer:
+        """The optimizer with which to train the model."""
+        return self._optimizer
+
+    @property
+    @abstractmethod
+    def model(self) -> DeepGP:
+        """The underlying GPflux model."""
+
+    @property
+    @abstractmethod
+    def keras_model(self) -> tf.keras.Model:
+        """Returns the compiled Keras model for training."""
+
+    def predict(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """Note: unless otherwise noted, this returns the mean and variance of the last layer
+        conditioned on one sample from the previous layers."""
+        return self.model.predict_f(query_points)
+
+    def predict_joint(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        raise NotImplementedError("Joint prediction not implemented for deep GPs")
+
+    @abstractmethod
+    def sample(self, query_points: TensorType, num_samples: int) -> TensorType:
+        raise NotImplementedError
+
+    def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        """Note: unless otherwise noted, this will return the prediction conditioned on one sample
+        from the lower layers."""
+        f_mean, f_var = self.model.predict_f(query_points)
+        return self.model.likelihood_layer.likelihood.predict_mean_and_var(f_mean, f_var)
+
+    def get_observation_noise(self) -> TensorType:
+        """
+        Return the variance of observation noise for homoscedastic likelihoods.
+        :return: The observation noise.
+        :raise NotImplementedError: If the model does not have a homoscedastic likelihood.
+        """
+        try:
+            noise_variance = self.model.likelihood_layer.likelihood.variance
+        except AttributeError:
+            raise NotImplementedError(f"Model {self!r} does not have scalar observation noise")
+
+        return noise_variance
+
+    def optimize(self, dataset: Dataset) -> None:
+        """
+        Optimize the model with the specified `dataset`.
+
+        :param dataset: The data with which to optimize the `model`.
+        """
+        self.keras_model.fit(
+            {"inputs": dataset.query_points, "targets": dataset.observations},
+            verbose=0,
```

Comment:
if we pass fit arguments directly to the class then we can simply unpack the dictionary here, no need for this separated treatment of epochs and batch_size